optuna/study/study.py

```diff
@@ -945,6 +949,47 @@ def objective(trial):
         for trial in trials:
             self.add_trial(trial)
 
+    def set_metric_names(self, metric_names: List[str]) -> None:
+        """Set metric names.
+
+        Each dimension of the objective function is named for each element of the
+        `metric_names` argument.
+
+        Example:
+
+            .. testcode::
+
+                import optuna
+                import pandas
+
+
+                def objective(trial):
+                    x = trial.suggest_float("x", 0, 10)
+                    return x**2, x + 1
+
+
+                study = optuna.create_study(directions=["minimize", "minimize"])
+                study.set_metric_names(["x**2", "x+1"])
+                study.optimize(objective, n_trials=3)
+
+                df = study.trials_dataframe()
+                assert isinstance(df, pandas.DataFrame)
+                assert list(df.values.keys()) == ["x**2", "x+1"]
+
+        .. seealso::
+            The names set by this method are used in :meth:`~optuna.study.Study.trials_dataframe`
+            and :func:`~optuna.visualization.plot_pareto_front`.
+
+        Args:
+            metric_names: A list of metric names for the objective function.
+        """
+        if len(self.directions) != len(metric_names):
+            raise ValueError(
+                "The number of objectives must match thhe length of the metric names."
+            )
+
+        self._storage.set_study_system_attr(self._study_id, "metric_names", metric_names)
```

Comment:
[question] Do you think we need to validate the length of `metric_names`? Or is that overkill?

Reply:
I think it would be better to validate here, since it is redundant handling the difference of length in `Study.trial_dataframe` or `plot_pareto_front`.

Reply:
Ah, I mean that I was concerned about the following case since the JSON string must be less than 1024 characters when using RDBStorage.

```python
study.set_metric_names(["String longer than 1024 characters....", "..."])
```

Reply:
Oh, I misunderstood. Sorry for the confusion.

If you want to perform such validation, I think it should be done on the set_system_attr side, not here. This is because not only set_metrics_name, but all methods that write strings to the RDB have restrictions on the strings to be written.

Reply:
I see. Sounds reasonable. Let me resolve this conversation.